spec/log-window-tests.js

```diff
@@ -54,59 +67,79 @@ describe('Log window', () => {
-        'log.html', jasmine.any(String), jasmine.any(String));
+      expect(chrome.windows.create).toHaveBeenCalledWith(new Object({
+        url: 'log.html', type: 'popup', height: 600, width: 700
+      }));
     });
 
-    it('reports the logging window is open', () => {
-      mockWindow.closed = false;
-      EmeLogWindow.instance.open();
+    it('reports the logging window is open', async function (done) {
+      expect(EmeLogWindow.instance.isOpen()).toBe(false);
+      await EmeLogWindow.instance.open();
       expect(EmeLogWindow.instance.isOpen()).toBe(true);
+      done();
     });
 
-    it('reports the logging window is closed', () => {
-      mockWindow.closed = true;
-      EmeLogWindow.instance.open();
+    it('reports the logging window is closed', async function (done) {
+      await EmeLogWindow.instance.open();
+      expect(EmeLogWindow.instance.isOpen()).toBe(true);
+      EmeLogWindow.instance.close(mockWindow.id);
       expect(EmeLogWindow.instance.isOpen()).toBe(false);
+      done();
     });
   });
 
+  it('appends a text log', async function (done) {
+    await EmeLogWindow.instance.open();
+    EmeLogWindow.instance.appendLog('text log 1');
+    expect(EmeLogWindow.instance.getTextLogs()).toEqual('text log 1');
+    done();
+  });
+
+  it('clears the text log', async function (done) {
+    await EmeLogWindow.instance.open();
+    EmeLogWindow.instance.appendLog('text log 2');
+    expect(EmeLogWindow.instance.getTextLogs()).toEqual(jasmine.any(String));
+    EmeLogWindow.instance.clear();
+    expect(EmeLogWindow.instance.getTextLogs()).toEqual('');
+    done();
+  });
+
   it('logs with timestamps', () => {
     const date = new Date('July 20, 1969 12:34:56 UTC');
     EmeLogWindow.instance.open();
-    EmeLogWindow.instance.appendLog({
+    EmeLogWindow.instance.appendLog(appendLog({
       timestamp: date.getTime(),
-    });
+    }));
     // Times are in the local user's timezone.  Without mocking that somehow,
     // we can only set expectations on the date format.
     expect(mockLogElement.querySelector('h3').textContent)
-        .toContain('Sun Jul 20 1969');
+      .toContain('Sun Jul 20 1969');
   });
 
   it('logs with durations', () => {
     EmeLogWindow.instance.open();
-    EmeLogWindow.instance.appendLog({
+    EmeLogWindow.instance.appendLog(appendLog({
       timestamp: Date.now(),
       duration: 15,
-    });
+    }));
     expect(mockLogElement.querySelector('h3').textContent)
-        .toContain('duration: 15.0 ms');
+      .toContain('duration: 15.0 ms');
   });
 
   it('shows warnings', () => {
     EmeLogWindow.instance.open();
-    EmeLogWindow.instance.appendLog({
+    EmeLogWindow.instance.appendLog(appendLog({
       timestamp: Date.now(),
       type: TraceAnything.LogTypes.Warning,
       message: 'Oh no!',
-    });
+    }));
 
     expect(mockLogElement.querySelector('.title').textContent)
-        .toContain('WARNING');
+      .toContain('WARNING');
```

Comment:
please revert these other style changes, too

Reply:
My VSCode with clangd for Chromium automatically removes 2 spaces when saving but reverted to the previous one.